crates/rpcsrv/src/proxy.rs

```diff
@@ -229,3 +245,31 @@ impl<A: ProxyAuthenticator + 'static> service::execution_service_server::Executi
         client.close_session(request).await
     }
 }
+
+/// Adapater stream for proxying streaming requests.
+struct ProxiedRequestStream<M> {
+    inner: Streaming<M>,
+}
+
+impl<M> ProxiedRequestStream<M> {
+    fn new(request: Streaming<M>) -> Self {
+        Self { inner: request }
+    }
+}
+
+impl<M> Stream for ProxiedRequestStream<M> {
+    type Item = M;
+    fn poll_next(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
+        match self.inner.poll_next_unpin(cx) {
+            Poll::Ready(Some(Ok(m))) => Poll::Ready(Some(m)),
+            Poll::Ready(Some(Err(e))) => {
+                // Don't know what we want to do yet, so just log and close down
+                // the stream.
+                warn!(%e, "received error when proxying request stream");
+                Poll::Ready(None)
+            }
```

Comment:
I'm not completely sure what we want to do here. The error would be a connection error, and so returning `Poll::Ready(None)` to end the stream seems reasonable right now.